content/Building an Engine-modularized API in Rails - Guillermo Leopold.md

```diff
@@ -0,0 +1,154 @@
+To explain this better, let’s take the orders engine as an example:
+  * The **order** engine has a model called **purchase**, it represents all the purchases in the system.
+  * Each **purchase** belongs to a **customer** (which belongs to the **user** engine) and a **customer** can have many **purchases** in terms of business.
+
+In order to be able to model this properly, we need to add a foreign key to the customer table in the purchases table and because of this, the **purchase** (and so too the **orders**  engine) will, by definition, need to have access to the **user** engine.
+This implies two considerations in terms of code:
+  1. We can define the `belongs_to` association in the purchase model due to orders engine having access to the user engine.
+  2. We won’t be able to have the `has_many` association in the user engine and because of that, if we need to have all the purchases for a user, we will need to define a scope in the purchases model that retrieves that information.
+
+
+## Building an engine
+
+After going through the first steps, the benefits of working with an API built with engines are already visible. The modules make it easier to split the team and start working on different features at the same time (let’s say: a pair of devs in the catalog, another pair in the admin and a few ones in the user engine).
+
+But it also made us feel uncomfortable as the lack of information, especially in terms of configuration and how to work between engines were a constant issue during the first few weeks.
+
+We decided to implement some scripts that would help us when we created new engines and we also wrote down some guides. At this point the collaboration between our development team was really important to learn how to deal with the engines.
+
+Having said that, the following is a helpful step-by-step guide on what to do when adding a new engine.
+
+The first step, of course, is to create the engine. To achieve that, rails provides a command with some flags:
+```rails plugin new engines/engine_name --mountable --skip-test --dummy-path=engines/engine_name/spec/dummy```.
+Let’s got a bit deeper on how this command is built:
+  * The ```--mountable``` flag means that we want to create the engine inside an isolated namespace.
+  * The ```--skip-test``` flag means that we don’t want the engine to include the scaffold for the testing suite as we are going to install rspec.
+  * The ```--dummy-path``` is used to define the path where we want to generate the dummy app that "works as the main app" when running the test suite.
+
+Following this, we started by cleaning our engine, removing a lot of unnecessary files, adding some common configuration and copying some files that we need across all the engines.
+
+After building a few engines, we decided to improve our experience and create our own script to create the engine and also complete the following forementioned steps.
```

Comment:
Can you add more info about this script? Maybe a minor example with the more basics things. 
An example plus the summary you provide below could be a good start point for someone that want to use engines in his projects